### ISSUE TITLE
evaluate_suite accepts duplicate task names and silently re-weights the rule-discovery search fitness (task axis of #127)

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#325]
## Summary

`evaluate_suite` (`alberta_framework/benchmarks/rule_discovery.py`) is documented as "Mean accuracy across the named micro tasks" and is the search fitness `run_search` optimizes ("Search fitness reads only `task_names`"). It loops `for name in task_names: per_task[name] = evaluate_population(...)` and returns `np.mean(np.stack([per_task[name] for name in task_names]), axis=0)`. A repeated task name is evaluated twice, stored once, and counted twice — the equal-weight task mean silently becomes a weighted objective.

`run_search` validates every neighbouring axis before compute — `require_unique_jax_seeds` on `eval_seeds` and `holdout_seeds` (#127 / PR #136 fixed exactly this defect on the seed axis of the same call path), task/holdout disjointness, seed disjointness — but never checks that `task_names` or `holdout_names` are sets, and the CLI's `--tasks` / `--holdout-tasks` (`nargs="+"`) accept repeats. `settings.task_names` in the result JSON records the duplicate, and the returned `per_task` dict has one entry per distinct name, so the artifact cannot be audited after the fact.

## Falsifier (deterministic, pristine `main` @ `bae17ee`)

```python
import jax.numpy as jnp, numpy as np
import alberta_framework.benchmarks.rule_discovery as rd

genomes = jnp.zeros((2, rd.GENOME_SIZE), dtype=jnp.float32)
values = {"M1": 0.05, "M2": 0.175}
rd.evaluate_population = lambda g, config, **kw: np.full((g.shape[0],), values[
    next(k for k, v in rd.MICRO_SUITE.items() if v is config)])
mean, per_task = rd.evaluate_suite(genomes, ("M1", "M1", "M2"), seeds=(0,))
print(mean[0], sorted(per_task))
print("equal-weight mean(M1, M2) =", (0.05 + 0.175) / 2, " 2/3*M1 + 1/3*M2 =", 2/3*0.05 + 1/3*0.175)
```

Output on `main` (also reproduced without monkeypatching on the real M1/M2 streams — 0.091667 vs 0.112500):

```
0.09166666666666667 ['M1', 'M2']
equal-weight mean(M1, M2) = 0.1125  2/3*M1 + 1/3*M2 = 0.09166666666666667
```

And end-to-end: `rule_discovery search --tasks M1 M1 M2 --holdout-tasks M1p …` is accepted and writes `settings.task_names = ["M1", "M1", "M2"]`, whereas `--eval-seeds 0 0` is refused before any genome is generated.

Expected: `evaluate_suite` and `run_search` refuse repeated task names (`task_names must contain unique task names`, mirroring the seed guard) before any evaluation.

## Dedup

Adjacent, closed: #127 / PR #136 (duplicate **seeds** in `evaluate_population`), #25 (duplicate experiment names), #152/#153 (duplicate control shards). No issue or PR mentions `evaluate_suite`, `task_names`, or duplicate tasks; no open PR touches `rule_discovery.py`.

Fix PR follows (failing-test-first).
